service/history/replication/task_fetcher.go

```diff
@@ -22,13 +22,15 @@ package replication
 
 import (
 	"context"
+	"sync"
```

Comment:
not related to this PR but matching tests have been flaky. is that in your radar?
```

panic: Log in goroutine after TestIntegrationSuite has completed: 2024-05-10T19:40:53.182Z	WARN	Async task dispatch timed out	{"component": "matching-engine", "wf-domain-name": "integration-test-domain-68911bf2-54c0-4d7e-952b-14281470170a", "wf-task-list-name": "integration-signal-with-start-workflow-id-reuse-test-tasklist", "wf-task-list-type": 0, "isolation-group": "", "wf-run-id": "6cc96716-87d3-4b1c-b6c9-343761bba38c", "wf-id": "integration-signal-with-start-workflow-id-reuse-test", "queue-task-id": 1, "error": "failed to offer task: context deadline exceeded", "wf-domain-id": "8452aa3e-f142-431f-8789-d32d2d758bd2", "logging-call-at": "taskReader.go:432"}
--
  |  
  | goroutine 74526 [running]:
  | testing.(*common).logDepth(0xc004d48820, {0xc00773e280, 0x259}, 0x3)
  | /usr/local/go/src/testing/testing.go:1003 +0x68c
  | testing.(*common).log(...)
  | /usr/local/go/src/testing/testing.go:985
  | testing.(*common).Logf(0xc004d48820, {0x49ccb9c, 0x2}, {0xc005fc6320, 0x1, 0x1})
  | /usr/local/go/src/testing/testing.go:1036 +0xa5
  | go.uber.org/zap/zaptest.testingWriter.Write({{0x61405a0?, 0xc004d48820?}, 0x0?}, {0xc0019cfc00, 0x25a, 0x400})
  | /go/pkg/mod/go.uber.org/zap@v1.13.0/zaptest/logger.go:130 +0x12d
  | go.uber.org/zap/zapcore.(*ioCore).Write(0xc00531aa80, {0x1, {0xc187bbf14ae53948, 0x9fe8afb6c4, 0x7c9b9e0}, {0x0, 0x0}, {0x4a09652, 0x1d}, {0x0, ...}, ...}, ...)
  | /go/pkg/mod/go.uber.org/zap@v1.13.0/zapcore/core.go:90 +0x19a
  | go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0077231e0, {0xc007734300, 0x7, 0xc})
  | /go/pkg/mod/go.uber.org/zap@v1.13.0/zapcore/entry.go:216 +0x2ed
  | go.uber.org/zap.(*Logger).Warn(0xc00871ed68?, {0x4a09652, 0x1d}, {0xc007734300, 0x7, 0xc})
  | /go/pkg/mod/go.uber.org/zap@v1.13.0/logger.go:195 +0x6e
  | github.com/uber/cadence/common/log/loggerimpl.(*loggerImpl).Warn(0xc00871ed68, {0x4a09652, 0x1d}, {0xc002c2e300, 0x6, 0x6})
  | /cadence/common/log/loggerimpl/logger.go:144 +0x18f
  | github.com/uber/cadence/service/matching.(*taskReader).dispatchSingleTaskFromBuffer(0xc0049fd8c0, {0x0, 0x0}, 0xc00757fe80)
  | /cadence/service/matching/taskReader.go:432 +0x14dd
  | github.com/uber/cadence/service/matching.(*taskReader).dispatchSingleTaskFromBufferWithRetries(...)
  | /cadence/service/matching/taskReader.go:403
  | github.com/uber/cadence/service/matching.(*taskReader).dispatchBufferedTasks(0xc0049fd8c0, {0x0, 0x0})
  | /cadence/service/matching/taskReader.go:173 +0x2c5
  | github.com/uber/cadence/service/matching.(*taskReader).Start.func1()
  | /cadence/service/matching/taskReader.go:134 +0xb3
  | created by github.com/uber/cadence/service/matching.(*taskReader).Start
  | /cadence/service/matching/taskReader.go:132 +0x171
  | FAIL	github.com/uber/cadence/host	686.924s
```

Reply:
yeah, I'll take a look at that after refactoring.